rollup.utils.mjs

```diff
@@ -40,32 +42,86 @@ const basePlugins = () => [
   dynamicImportVars({}),
 ];
 
-const baseConfig = (baseDir) => {
-  const outDir = path.resolve(baseDir, 'dist');
+const isInput = (id, input) => {
+  if (typeof input === 'string') {
+    return id.includes(path.resolve(input));
+  }
+
+  return Object.values(input).some((i) => id.includes(path.resolve(i)));
+};
+
+const baseConfig = (opts = {}) => {
+  const { rootDir, outDir = './dist', input = './src/index.ts', ...rest } = opts;
 
   return defineConfig({
-    input: path.resolve(baseDir, 'src/index.ts'),
-    external: (id) => !path.isAbsolute(id) && !id.startsWith('.'),
-    output: [
-      {
-        dir: outDir,
-        entryFileNames: '[name].js',
-        chunkFileNames: 'chunks/[name]-[hash].js',
-        exports: 'auto',
-        format: 'cjs',
-        sourcemap: true,
+    input,
+    external: isExernal,
+    output: baseOutput({ outDir, rootDir }),
+    plugins: basePlugins(),
+    onwarn(warning, warn) {
+      if (warning.code === 'MIXED_EXPORTS') {
+        // json files are always mixed exports
+        if (warning?.id?.endsWith('.json')) {
+          return;
+        }
+
+        // we only care about mixed exports in our input files
+        if (warning.id && !isInput(warning.id, input)) {
+          return;
+        }
```

Comment:
what are input files in this context? files to be built?

Reply:
this is the entry points we build which are basically the public entrypoints :)